diff --git a/modules/eddl/include/ecvl/augmentations.h b/modules/eddl/include/ecvl/augmentations.h
index f09675e..ed490ed 100644
--- a/modules/eddl/include/ecvl/augmentations.h
+++ b/modules/eddl/include/ecvl/augmentations.h
@@ -201,6 +201,8 @@ public:
     template<typename ...Ts>
     SequentialAugmentationContainer(Ts&&... t) : augs_({ std::make_shared<Ts>(std::forward<Ts>(t))... }) {}
 
+    SequentialAugmentationContainer(std::vector<std::shared_ptr<Augmentation>> augs) : augs_(augs) {}
+
 	SequentialAugmentationContainer(std::istream& is) {
 		while (true) {
 			std::string name;
@@ -642,4 +644,4 @@ public:
 
 } // namespace ecvl
 
-#endif // AUGMENTATIONS_H_
\ No newline at end of file
+#endif // AUGMENTATIONS_H_
